pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -523,14 +523,25 @@ func handleLastRebootReason(ctx *nodeagentContext) {
 	}
 
 	agentlog.DiscardBootReason(log)
-	// Make sure we log the reboot stack
+	// Make sure we log the reboot stack or dmesg
 	if len(rebootStack) > 0 {
-		log.Warnf("Found RebootStack %d bytes", len(rebootStack))
 		lines := strings.Split(rebootStack, "\n")
-		log.Warnf("Found RebootStack %d bytes %d lines",
-			len(rebootStack), len(lines))
-		for i, l := range lines {
-			log.Warnf("[%d]: %s", i, l)
+
+		if bootReason == types.BootReasonKernel {
+			// Kernel crashes are a bit special, since file contains not
+			// only a backtrace, but the whole dmesg of a crashed kernel.
+			// Mark each line with 'dmesg' prefix for easy grep.
+			log.Warnf("Found dmesg %d bytes %d lines",
+				len(rebootStack), len(lines))
+			for i, l := range lines {
+				log.Warnf("[dmesg %d]: %s", i, l)
+			}
+		} else {
+			log.Warnf("Found RebootStack %d bytes %d lines",
```

Comment:
I don't like this stupid "if branch" which differs only by the message we print, but I did not want to change the original message, since I do not know how many people use it for debugging nowadays, so just don't want to break current procedure. But if no one cares  - I'm happy to introduce simple:
```
if bootReason == BootReasonKernel:
      prefix = "dmesg"
else:
      prefix = "stack"
```

or similar

Reply:
@eriknordmark ^^^

Reply:
Works for me.